Configurations/unix-Makefile.tmpl

```diff
@@ -407,10 +407,10 @@ LIB_CPPFLAGS={- our $lib_cppflags =
                           @{$config{lib_cppflags}},
                           @{$config{shared_cppflag}});
                 join(' ', $lib_cppflags,
-                          (map { '-D'.$_ }
-                               'OPENSSLDIR="\"$(OPENSSLDIR)\""',
-                               'ENGINESDIR="\"$(ENGINESDIR)\""',
-                               'MODULESDIR="\"$(MODULESDIR)\""'),
+                          (map { '"-D'.$_.'"' }
+                               'OPENSSLDIR=\"$(OPENSSLDIR)\"',
+                               'ENGINESDIR=\"$(ENGINESDIR)\"',
+                               'MODULESDIR=\"$(MODULESDIR)\"'),
```

Comment:
What if there are spaces in any of these DIR variables?

Reply:
That's fine, they are included by the outer quotes

Reply:
and why do they need to be escaped at all?

Reply:
Because they'll be joined into a string that ends up on the command line. The result will look like this:

```
... "-DOPENSSLDIR=\"/usr/local/ssl\"" ...
```

The escaped quotes need to be there, because the uses of these macros in our code are expecting the values to be strings

Reply:
I wonder why we're having this sudden discussion, about something that's been around for ages. It's really just moving the starting quote. For reference, see the issue that's pointed at in this PRs description.

Reply:
My mistake, I've overlooked the quotes are moved, not dropped.

Reply:
Ah sorry for the noise, but I still wonder if it would be better to use something like `quotify1('-D'.$_)`
just in case there are other unquoted special characters in the string, like `$"\` and/or the infamous back-ticks.